cmd/commandfuncs.go

```diff
@@ -700,3 +704,25 @@ func apiRequest(adminAddr, method, uri string, body io.Reader) error {
 
 	return nil
 }
+
+func sdNotify(state string) error {
+	socketAddr := &net.UnixAddr{
+		Name: os.Getenv("NOTIFY_SOCKET"),
+		Net:  "unixgram",
+	}
+
+	if socketAddr.Name == "" {
+		return nil
+	}
+
+	conn, err := net.DialUnix(socketAddr.Net, nil, socketAddr)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	if _, err := conn.Write([]byte(state)); err != nil {
```

Comment:
Shouldn't this be a `io.Copy(conn, strings.NewReader(state))` instead?

Reply:
Why's that?

Reply:
`conn.Write()` does not guarantee that it will write all the data. `io.Copy()` does. I use this consistently in go code I write as it avoids bugs which are hard to debug.

Reply:
That's a good point.

Reply:
maybe not required here, but consistency counts IMO.

Reply:
Updated.